Mixpanel/UIView+MPHelpers.m

```diff
@@ -0,0 +1,231 @@
+        CGContextSetAllowsAntialiasing(context, NO);
+        CGContextClearRect(context, CGRectMake(0, 0, 8, 8));
+        CGContextSetInterpolationQuality(context, kCGInterpolationNone);
+        CGContextDrawImage(context, CGRectMake(0,0,8,8), [originalImage CGImage]);
+        CGColorSpaceRelease(space);
+        CGContextRelease(context);
+        for(int i = 0; i < 32; i++) {
+            int j = 2*i;
+            int k = 2*i + 1;
+            data4[i] = (((data32[j] & 0x80000000) >> 24) | ((data32[j] & 0x800000) >> 17) | ((data32[j] & 0x8000) >> 10) | ((data32[j] & 0x80) >> 3) |
+                        ((data32[k] & 0x80000000) >> 28) | ((data32[k] & 0x800000) >> 21) | ((data32[k] & 0x8000) >> 14) | ((data32[k] & 0x80) >> 7));
+        }
+        result = [[NSData dataWithBytes:data4 length:32] base64EncodedStringWithOptions:0];
+    }
+    return result;
+}
+
+- (NSArray *)mp_encryptedTargetActions
+{
+    NSArray *targetActions = [self mp_targetActions];
+    NSMutableArray *encryptedActions = [NSMutableArray array];
+    for (int i = 0 ; i < [targetActions count]; i++) {
+        [encryptedActions addObject:mp_encryptHelper(targetActions[i])];
+    }
+    return encryptedActions;
+}
+
+- (NSString *)mp_text
+{
+    NSString *text = nil;
+    SEL titleSelector = NSSelectorFromString(@"title");
+    if ([self isKindOfClass:[UILabel class]]) {
+        text = ((UILabel *)self).text;
+    } else if ([self isKindOfClass:[UIButton class]]) {
+        text = [((UIButton *)self) titleForState:UIControlStateNormal];
+    } else if ([self respondsToSelector:titleSelector]) {
+        IMP titleImp = [self methodForSelector:titleSelector];
+        void *(*func)(id, SEL) = (void *(*)(id, SEL))titleImp;
+        id title = (__bridge id)func(self, titleSelector);
+        if ([title isKindOfClass:[NSString class]]) {
+            text = title;
+        }
+    }
+    return text;
+}
+        
+static NSString* mp_encryptHelper(id input)
+{
+    NSString *SALT = @"1l0v3c4a8s4n018cl3d93kxled3kcle3j19384jdo2dk3";
+    NSMutableString *encryptedStuff = nil;
+    if ([input isKindOfClass:[NSString class]]) {
+        NSData *data = [[input stringByAppendingString:SALT]  dataUsingEncoding:NSASCIIStringEncoding];
+        uint8_t digest[CC_SHA256_DIGEST_LENGTH];
+        CC_SHA256(data.bytes, data.length, digest);
+        encryptedStuff = [NSMutableString stringWithCapacity:CC_SHA256_DIGEST_LENGTH * 2];
+        for (int i = 0; i < CC_SHA1_DIGEST_LENGTH; i++) {
+            [encryptedStuff appendFormat:@"%02x", digest[i]];
+        }
+    }
+    return encryptedStuff;
+}
+
+#pragma mark - Aliases for compatibility
+- (NSString *)mp_varA
+{
+    return mp_encryptHelper([self mp_viewId]);
+}
+
+- (NSString *)mp_varB
+{
+    return mp_encryptHelper([self mp_controllerVariable]);
+}
+
+- (NSString *)mp_varC
+{
+    return mp_encryptHelper([self mp_imageFingerprint]);
+}
+
+- (NSArray *)mp_varSetD
+{
+    return [self mp_encryptedTargetActions];
```

Comment:
get rid of `mp_encryptedTargetActions` and just inline the code here, so we only have `mp_varSetD` and `mp_targetActions`